pkgs/development/python-modules/python-troveclient/default.nix

```diff
@@ -0,0 +1,87 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  httplib2,
+  keystoneauth1,
+  openstackdocstheme,
+  osc-lib,
+  oslo-i18n,
+  oslo-utils,
+  pbr,
+  prettytable,
+  python-mistralclient,
+  python-openstackclient,
+  python-swiftclient,
+  pythonOlder,
+  requests-mock,
+  requests,
+  setuptools,
+  sphinxcontrib-apidoc,
+  sphinxHook,
+  stestr,
+}:
+
+buildPythonPackage rec {
+  pname = "python-troveclient";
+  version = "8.5.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "openstack";
+    repo = "python-troveclient";
+    rev = "refs/tags/${version}";
+    hash = "sha256-lfnAmQ/IxEdc+XxC0dYxK2FgY7csNewGPuQuq0dNffM=";
+  };
+
+  env.PBR_VERSION = version;
+
+  nativeBuildInputs = [
+    openstackdocstheme
+    sphinxHook
+    sphinxcontrib-apidoc
+  ];
+
+  sphinxBuilders = [ "man" ];
+
+  build-system = [
+    pbr
+    setuptools
+  ];
+
+  dependencies = [
+    keystoneauth1
+    osc-lib
+    oslo-i18n
+    oslo-utils
+    prettytable
+    python-mistralclient
+    python-openstackclient
```

Comment:
Note for reviewers: I cannot add `python-troveclient` to `openstackclient-full` (python-openstackclient) otherwise it would cause an infinite recursion issue.